frontend/cs/r1cs/api.go

```diff
@@ -687,6 +688,19 @@ func (builder *builder) Compiler() frontend.Compiler {
 
 func (builder *builder) Commit(v ...frontend.Variable) (frontend.Variable, error) {
 
+	// add a random mask to v
+	{
```

Comment:
As far as I understand, doing it this way adds a randomized value every time we call a commit? But in case we call it many times it could add a lot of redundant constraints/vars.

Reply:
Aah, I realized that `Commit` always adds a commitment indeed and we have to use multicommit anyway. So no issue here.